miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppManifestVerifier.kt

```diff
@@ -0,0 +1,72 @@
+package com.rakuten.tech.mobile.miniapp
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.util.Base64
+import android.util.Log
+import androidx.annotation.VisibleForTesting
+import androidx.security.crypto.EncryptedSharedPreferences
+import androidx.security.crypto.MasterKeys
+import com.rakuten.tech.mobile.miniapp.storage.CachedManifest
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.withContext
+import java.security.MessageDigest
+
+internal class MiniAppManifestVerifier
```

Comment:
This class is very similar to `CachedMiniAppVerifier`. I think you could share the functionality between them either by creating a base class or something like `StoredHashVerifier` which contains common functionality.

Reply:
Yeah, I was concerned the same, I'll update this part accordingly.